cmd/kubeadm/app/util/chroot_unix.go

```diff
@@ -0,0 +1,46 @@
+// +build !windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+// MaybeChroot chroot()s to the new path, if non-empty.
+// NB: All file paths after this call are effectively relative to
+// `rootfs`
+func MaybeChroot(rootfs string) error {
+	if rootfs == "" {
```

Comment:
also, i think it's a good idea to add a unit test for the unix case of this function.
```
chroot_test.go

// +build !windows
...

func ChrootTest(...)

```
please, let me know if you don't have the time to make these amends.

Reply:
The test will require root (or CAP_SYS_ADMIN) in order to execute chroot().  Are our automated tests run as root? (I hope not..)

Reply:
that's a good point. they are not!
we shouldn't test the `Chroot()/Chdir()` calls in that case.

but this brings a related topic - `rootfs` is added as a persistent flag for all kubeadm commands, while only `kubead init/join` require root.
i think we should add the flag only for these commands, as something like `kubeadm config ...` or `kubeadm version` would not benefit from that.

WDYT?
also /cc @timothysc

Reply:
I added a test.  Tbh, given it only works when invoked as root, I'm not sure it's worth the additional complexity.